src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,81 @@
+package duke;
+
+import command.Command;
+import command.FindCommand;
+import command.ListCommand;
+import command.DeleteCommand;
+import command.DoneCommand;
+import command.AddCommand;
+import command.EndCommand;
+
+import task.Deadline;
+import task.Todo;
+import task.Event;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+/**
+ * Parses the command string from the user and check its validity.
+ */
+public class Parser {
+    /**
+     * Parse the command string given.
+     * @param command Command string.
+     * @return Command object.
+     * @throws DukeException Invalid actions / missing values.
+     * @throws ParseException Invalid variable to parse.
+     */
+    public static Command parse(String command) throws DukeException, ParseException {
+        String[] task = command.split(" ", 2);
```

Comment:
What a great idea to use split()'s limit parameter to split the command and the details! I forgot it even existed. Very clever. :+1: